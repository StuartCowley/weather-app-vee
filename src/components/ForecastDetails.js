import React from "react";
import PropTypes from "prop-types";
import moment from "moment";
import "../styles/ForecastDetails.css";

function ForecastDetails({ forecast }) {
  const { date, humidity, temperature, wind } = forecast;
  const formattedDate = moment(date).format("dddd Do MMMM");
  return (
    <div className="forecast-details">
      <div className="forecast-details__date">{formattedDate}</div>
      <div className="forecast-details__temperature">
        Max Temperature: {temperature.max}&deg;C
      </div>
      <div className="forecast-details__temperature">
        Min Temperature: {temperature.min}&deg;C
      </div>
      <div className="forecast-details__humidity">Humidity: {humidity}</div>
      <div className="forecast-details__wind">
        Wind: {wind.speed}
        {wind.direction}
      </div>
    </div>
  );
}

export default ForecastDetails;

ForecastDetails.propTypes = {
  forecast: PropTypes.shape({
    date: PropTypes.number,
    temperature: PropTypes.shape({
      max: PropTypes.number,
      min: PropTypes.number,
    }),
    humidity: PropTypes.number,
    wind: PropTypes.shape({
      speed: PropTypes.number,
      direction: PropTypes.string,
    }),
  }).isRequired,
};
