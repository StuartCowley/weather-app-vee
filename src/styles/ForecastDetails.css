.forecast-details {
  margin-bottom: 1rem;
  font-size: 1.2rem;
  font-weight: bold;
  margin-top: 1.5rem;
  display: flex;
  flex-direction: column;
  align-items: center;
}

.forecast-details > div {
  margin: 0 0 0.2rem 0;
}

.forecast-details__date {
  font-weight: bold;
  font-size: 1.8rem;
}

.forecast-details__icon {
  font-size: 4rem;
  font-weight: bold;
}

.forecast-details__wind {
  white-space: pre-wrap;
}

@media (min-width: 992px) {
  .forecast-details {
    font-size: 1.3rem;
    font-weight: bold;
    margin-bottom: 1.5rem;
  }

  .forecast-details__date {
    font-weight: bold;
    font-size: 2rem;
  }

  .forecast-details__icon {
    font-size: 3rem;
    font-weight: bold;
  }
}
