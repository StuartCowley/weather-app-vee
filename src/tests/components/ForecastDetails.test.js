import React from "react";
import { render } from "@testing-library/react";
import ForecastDetails from "../../components/ForecastDetails";

describe("ForecastDetails", () => {
  const validProps = {
    date: 1111111,
    temperature: {
      min: 12,
      max: 22,
    },
    humidity: "80%",
    wind: {
      speed: "60mph",
      direction: "s",
    },
  };

  it("renders correctly", () => {
    const { asFragment } = render(
      <ForecastDetails
        date={validProps.date}
        temperature={validProps.temperature}
        humidity={validProps.humidity}
        wind={validProps.wind}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("it renders correct values for props", () => {
    const { getByText, getByTestId } = render(
      <ForecastDetails
        date={validProps.date}
        temperature={validProps.temperature}
        humidity={validProps.humidity}
        wind={validProps.wind}
      />
    );

    expect(getByText("Thu Jan 01 1970")).toHaveAttribute(
      "class",
      "forecast-details__date"
    );
    expect(getByText("22")).toHaveAttribute(
      "class",
      "forecast-details__temperature-max"
    );
    expect(getByText("12ËšC")).toHaveAttribute(
      "class",
      "forecast-details__temperature-min"
    );
    expect(getByText("80%")).toHaveAttribute(
      "class",
      "forecast-details__humidity"
    );
    expect(getByText("60mph")).toHaveAttribute(
      "class",
      "forecast-details__wind"
    );
    expect(getByText("s")).toHaveAttribute("class", "forecast-details__date");
  });
});
// TODO need to determine if getByText works for single letter. Icon / getByTestID needed?
