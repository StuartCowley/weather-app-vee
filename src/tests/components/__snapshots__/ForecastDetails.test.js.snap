// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastDetails renders correctly 1`] = `
<DocumentFragment>
  <div
    class="forecast-details"
  >
    <div
      class="forecast-details__date"
    >
      Thursday 1st January
    </div>
    <div
      class="forecast-details__temperature"
    >
      Max Temperature: 22°C
    </div>
    <div
      class="forecast-details__temperature"
    >
      Min Temperature: 12°C
    </div>
    <div
      class="forecast-details__humidity"
    >
      Humidity: 80
    </div>
    <div
      class="forecast-details__wind"
    >
      Wind: 60ene
    </div>
  </div>
</DocumentFragment>
`;
